src/app/clusters/bindings/bindings.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Binding Server Cluster
+ ***************************************************************************/
+
+#include "af.h"
+
+#include <app/util/binding-table.h>
+#include <support/logging/CHIPLogging.h>
+
+using namespace chip;
+
+EmberStatus prepareBinding(EmberBindingTableEntry & binding, NodeId nodeId, GroupId groupId, EndpointId endpointId,
+                           ClusterId clusterId)
+{
+    if ((groupId && (nodeId != 0 || endpointId != 0)) || (nodeId && endpointId == 0))
```

Comment:
Why can't we bind to endpoint 0?

I would prefer that we either explicitly use `Optional` to represent not-passed things here, or have a struct that can represent "either group id or (node, endpoint) pair" or something...

Probably ok to land as-is for now to unblock things, but please file a followup on fixing this?

Reply:
> Why can't we bind to endpoint 0?
> 
> I would prefer that we either explicitly use `Optional` to represent not-passed things here, or have a struct that can represent "either group id or (node, endpoint) pair" or something...
> 
> Probably ok to land as-is for now to unblock things, but please file a followup on fixing this?

I have simplified the expression as `if(groupId && nodeId)` but I will file a followup to use a struct.